src/FSAL/FSAL_CORTXFS/mds.c

```diff
@@ -485,29 +485,30 @@ int cortxfs_pmds_ini(struct kvsfs_fsal_module *kvsfs,
 int cortxfs_pmds_fini(struct kvsfs_fsal_module *kvsfs)
 {
 	int rc = 0;
-
 	dassert(kvsfs != NULL);
+	
+	if (kvsfs->mds_ctx->pnfs_role > CORTXFS_PNFS_DS )
+	{	
+		rc = pthread_mutex_lock(&kvsfs->mds_ctx->mds_mutex);
+		if (rc != 0) {
+			LogCrit(COMPONENT_PNFS, "pthread_mutex_lock failed: %x", rc);
+			goto out;
+		}		
 
-	rc = pthread_mutex_lock(&kvsfs->mds_ctx->mds_mutex);
-	if (rc != 0) {
-		LogCrit(COMPONENT_PNFS, "pthread_mutex_lock failed: %x", rc);
-		goto out;
-	}
-
-	cortxfs_pmds_cleanup(kvsfs->mds_ctx->mds);
-
-	rc = pthread_mutex_unlock(&kvsfs->mds_ctx->mds_mutex);
-	if (rc != 0) {
-		LogCrit(COMPONENT_PNFS, "pthread_mutex_unlock failed: %x", rc);
-		goto out;
-	}
+		cortxfs_pmds_cleanup(kvsfs->mds_ctx->mds);
 
+		rc = pthread_mutex_unlock(&kvsfs->mds_ctx->mds_mutex);
+		if (rc != 0) {
+			LogCrit(COMPONENT_PNFS, "pthread_mutex_unlock failed: %x", rc);
+			goto out;
+		}	
+	
+  	}
 	rc = pthread_mutex_destroy(&kvsfs->mds_ctx->mds_mutex);
```

Comment:
Pls put this mutex destroy code inside the if block from line number 490. Rest looks good.